decidim-bulletin_board-js/src/voter/voter.js

```diff
@@ -48,26 +48,55 @@ export class Voter {
    * Encrypts the data using the wrapper.
    *
    * @param {Object} data - The data to be encrypted.
-   * @returns {Promise<Object>} - The data encrypted and its hash.
+   * @returns {Promise<Object>} - The data encrypted.
    */
   async encrypt(data) {
-    const encryptedVote = await this.wrapper.encrypt(data);
+    const ballot = await this.wrapper.encrypt(data);
+    return ballot;
+  }
 
+  /**
+   * Encrypts the data using the wrapper.
+   *
+   * @param {Object} encryptedVote - The encryptedVote to be encrypted.
+   * @returns {Promise<Object>} - The data encrypted and its hash.
+   */
+  async encryptBallot(encryptedVote) {
+    const encryptedBallot = encryptedVote.encryptedBallot;
     return window.crypto.subtle
-      .digest("SHA-256", new TextEncoder().encode(encryptedVote))
+      .digest("SHA-256", new TextEncoder().encode(encryptedBallot))
       .then((hashBuffer) => {
         const hashArray = Array.from(new Uint8Array(hashBuffer));
         const hashHex = hashArray
           .map((b) => b.toString(16).padStart(2, "0"))
           .join("");
 
         return {
-          encryptedVote,
+          encryptedVote: encryptedBallot,
```

Comment:
You should also rename the key to `encryptedBallot` and since the `hashHex` is from the ballot you can also use `encryptedBallotHash`. Then you will need to update the rest of the code accordingly.